testsuite/pytests/test_eprop_plasticity.py

```diff
@@ -0,0 +1,397 @@
+    dtype_in_spks = np.float32
+
+    input_spike_bools = (np.random.rand(steps["sequence"], n_in) < input_spike_prob).swapaxes(0, 1)
+
+    sequence_starts = np.arange(0.0, duration["task"], duration["sequence"]) + duration["offset_gen"]
+    params_gen_spk_in = []
+    for input_spike_bool in input_spike_bools:
+        input_spike_times = np.arange(0.0, duration["sequence"], duration["step"])[input_spike_bool]
+        input_spike_times_all = [input_spike_times + start for start in sequence_starts]
+        params_gen_spk_in.append({"spike_times": np.hstack(input_spike_times_all).astype(dtype_in_spks)})
+
+    nest.SetStatus(gen_spk_in, params_gen_spk_in)
+
+    # Create output
+
+    def generate_superimposed_sines(steps_sequence, periods):
+        n_sines = len(periods)
+
+        amplitudes = np.random.uniform(low=0.5, high=2.0, size=n_sines)
+        phases = np.random.uniform(low=0.0, high=2.0 * np.pi, size=n_sines)
+
+        sines = [
+            A * np.sin(np.linspace(phi, phi + 2.0 * np.pi * (steps_sequence // T), steps_sequence))
+            for A, phi, T in zip(amplitudes, phases, periods)
+        ]
+
+        superposition = sum(sines)
+        superposition -= superposition[0]
+        superposition /= max(np.abs(superposition).max(), 1e-6)
+        return superposition
+
+    target_signal = generate_superimposed_sines(steps["sequence"], [1000, 500, 333, 200])
+
+    params_gen_rate_target = {
+        "amplitude_times": np.arange(0.0, duration["task"], duration["step"]) + duration["total_offset"],
+        "amplitude_values": np.tile(target_signal, n_iter * group_size),
+    }
+
+    nest.SetStatus(gen_rate_target, params_gen_rate_target)
+
+    # Create learning window
+
+    params_gen_learning_window = {
+        "amplitude_times": [duration["total_offset"]],
+        "amplitude_values": [1.0],
+    }
+
+    nest.SetStatus(gen_learning_window, params_gen_learning_window)
+
+    # Simulate
+
+    nest.Simulate(duration["sim"])
+
+    # Read out recorders
+
+    events_mm_out = mm_out.get("events")
+
+    # Evaluate training error
+
+    readout_signal = events_mm_out["readout_signal"]
+    target_signal = events_mm_out["target_signal"]
+    senders = events_mm_out["senders"]
+
+    readout_signal = np.array([readout_signal[senders == i] for i in set(senders)])
+    target_signal = np.array([target_signal[senders == i] for i in set(senders)])
+
+    readout_signal = readout_signal.reshape((n_out, n_iter, group_size, steps["sequence"]))
+    target_signal = target_signal.reshape((n_out, n_iter, group_size, steps["sequence"]))
+
+    loss = 0.5 * np.mean(np.sum((readout_signal - target_signal) ** 2, axis=3), axis=(0, 2))
+
+    # Verify results
+
+    if neuron_model == "eprop_iaf":
+        loss_NEST_reference = np.array(
+            [
+                115.18883444998838,
+                102.72225734161191,
+                91.09860416985143,
+                81.78596643400449,
+                67.49056117141924,
```

Comment:
For me, the test was successful only when using this last loss. Could you please test this? If this depends on the machine, we need to adjust the tolerance.
```suggestion
                67.02284542820075,
```

Reply:
Do any of the other 2 cases fail?

Reply:
Does any of the other 2 cases fail?

Reply:
No, only that one.

Reply:
Fixed in [27da18c](https://github.com/JesusEV/nest-simulator/pull/24/commits/27da18c31d6f736ccce17c61b4afd158aa38d20c).